src/Avalonia.Controls/Templates/IDataTemplate.cs

```diff
@@ -6,21 +6,7 @@ namespace Avalonia.Controls.Templates
     /// <summary>
     /// Interface representing a template used to build a control for a piece of data.
     /// </summary>
-    public interface IDataTemplate : ITemplate<object, IControl>
+    public interface IDataTemplate : IDataTemplate<object, IControl>
```

Comment:
IDataTemplate still depends on the IControl. We need to be able to do something like this:

```xml
<ExportedMenu Items="{Binding MyITems}">
   <ExportedMenu.DataTemplates>
      <DataTemplate TargetType="{x:Type MyItemModel}">
           <ExportedMenuItem Text="{Binding Text}" Icon="{Binding Icon}"/>
      </DataTemplate>
   </ExportedMenu.DataTemplates>
</ExportedMenu>
```

Reply:
Ah ok, I had assumed the data template would have to be a different type, e.g. i dunno, `MenuDataTemplate`.

The problem with this is that `DataTemplate` when it's usually used _does_ need to create an `IControl`, so we can either have two different types or do the type checking at runtime. Both are pretty nasty, I'm not sure which i like the least...

Reply:
We might also somehow mark a property as "templates only", so TargetType would be assigned directly to `ExportedMenuItem` that could also improve developer experience in other areas, e. g.
```xml
<ListBox>
    <ListBox.DataTemplates>
          <TextBlock TargetType="{x:Type vm:MyModel}" Text="{Binding Text}"/>
    </ListBox.DataTemplates>
</ListBox>
```